src/main/java/org/opensearch/ad/feature/AbstractRetriever.java

```diff
@@ -14,15 +14,20 @@
 import java.util.Arrays;
 import java.util.Iterator;
 import java.util.List;
+import java.util.Locale;
+import java.util.Map;
 import java.util.Optional;
 
 import org.opensearch.ad.common.exception.EndRunException;
 import org.opensearch.search.aggregations.Aggregation;
+import org.opensearch.search.aggregations.AggregationBuilder;
 import org.opensearch.search.aggregations.Aggregations;
 import org.opensearch.search.aggregations.bucket.MultiBucketsAggregation;
+import org.opensearch.search.aggregations.bucket.composite.CompositeAggregationBuilder;
 import org.opensearch.search.aggregations.metrics.InternalTDigestPercentiles;
 import org.opensearch.search.aggregations.metrics.NumericMetricsAggregation.SingleValue;
 import org.opensearch.search.aggregations.metrics.Percentile;
+import org.opensearch.search.builder.SearchSourceBuilder;
 
 public class AbstractRetriever {
```

Comment:
Out of curiousity, why is this class called abstract but not actually abstract?

Reply:
good question, changed to make it abstract.